lisa/sut_orchestrator/azure/credential.py

```diff
@@ -71,6 +73,12 @@ def __post_init__(self) -> None:
         secret.add_secret(self.token)
 
 
+@dataclass_json()
+@dataclass
+class WorkloadIdentityCredentialSchema(AzureCredentialSchema):
+    allow_all_tenants: bool = False
```

Comment:
Move it to the base class. Others may need it too. After that the schema is not needed, the workload identity credential can use the base schema directly.